### PR TITLE
Warn about practices that lead to flaky tests

### PR DESCRIPTION
**Pull request by elharo**
@tdcmeehan

## Description
More rules for tests

## Motivation and Context
This has caused flaky tests in the past

## Impact
none

## Test Plan
CI

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -357,6 +357,14 @@ We recommend you use IntelliJ as your IDE. The code style template for the proje
 * Categorize errors when throwing an exception
 * **Tests**
     * Avoid adding `Thread.sleep` in tests--these can fail due to environmental conditions, such as garbage collection or noisy neighbors in the CI environment.
+    * Do not use random values in tests. All tests should be reproducible.
+    * Be extremely careful when storing test data in fields. This is really
```
Comment (steveburnett) [inline CONTRIBUTING.md:361]:
```suggestion
    * Be careful of storing test data in fields, because doing so is
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -357,6 +357,14 @@ We recommend you use IntelliJ as your IDE. The code style template for the proje
...
+      only appropriate for constant, immutable values. Most Presto tests
+      run with TestNG which, unlike JUnit, does not create a new object for
+      each test. Shared instance fields can make tests tightly coupled,
+      order dependent, and flaky. If you do use instance fields,
```
Comment (steveburnett) [inline CONTRIBUTING.md:365]:
```suggestion
      order dependent, and flaky. If you use instance fields, 
```

**Review by steveburnett — CHANGES_REQUESTED** [review]
Nice work, thanks! From this draft I understood what the problem is, why it is a problem, and how to avoid the problem. Suggested a couple of minor revisions to tighten the sentence, but nothing significant.

File: CONTRIBUTING.md (hunk 1 above)
Comment (elharo) [reply CONTRIBUTING.md]: done

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

**Pull request merged**